android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/dao/RegisterDaoFactory.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.data.local.register.dao
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.smartregister.fhircore.engine.appfeature.model.HealthModule
+import org.smartregister.fhircore.engine.domain.repository.RegisterDao
+
+@Singleton
+class RegisterDaoFactory
+@Inject
+constructor(
+  val ancPatientRegisterDao: AncPatientRegisterDao,
```

Comment:
We need a new issue to refactor this to a generic register that is configured cc @dubdabasoduba

Reply:
The decision to separate the data access objects was informed by the following:

1. **Ease of refactor** - Each register had its own repository for accessing the data. This has now been refactored to one register repository that delegates data access to the register DAOs. We use the '`HealthModule`' property to differentiate the register type.
2. **Isolation** implementation for the
3. **Reduce code complexity** by avoiding multiple `if/when` statements when using one generic register. Implementation for accessing/manipulating data per register differs and I do not think that can be made generic enough.

We can discuss this with @dubdabasoduba and perhaps come up with a better approach but I do not think data manipulation/access for all registers is something can we can make generic.

Reply:
The application treats ever `PatientManagement` & `HouseholdManagement` feature configured here as a register [config_app_feature](https://github.com/opensrp/fhircore/blob/implement-generic-app-register/android/quest/src/test/resources/configs/ecbis-saa/config_app_feature.json)